platalea/decoders.py

```diff
@@ -170,6 +170,7 @@ def beam_search(self, encoder_outputs, beam_size):
                 # Filter ended sequences
                 ended = (new_hyps[:, -1] == self.eos_id).nonzero()[0]
                 best_ended_idx = None
+                best_ended_score = -np.inf  # TODO: please review whether this initial value makes sense (need something, otherwise undefined in if statement below, except that best_ended_hyp is always set to none on line 134 above, but I guess that may change? If not, then the if statement is perhaps not useful and could be removed?)
```

Comment:
The check `best_ended_hyp is None` prevents evaluation of `new_scores[ended[0]] > best_ended_score` before `best_ended_score` is set but setting it to -inf is probably safest
```suggestion
                best_ended_score = -np.inf
```

Reply:
Yes indeed, it's never reached in the current setup, that's why Python never triggered an exception on this, but the linter did pick it up, apparently :)